server/app/services/question/QuestionServiceImpl.java

```diff
@@ -112,6 +116,12 @@ public void restoreQuestion(Long id) throws InvalidUpdateException {
     if (question.isEmpty()) {
       throw new InvalidUpdateException("Did not find question.");
     }
+    ActiveAndDraftQuestions activeQuestions =
+        readOnlyQuestionService().getActiveAndDraftQuestions();
+    if (activeQuestions.getDeletionStatus(question.get().getQuestionDefinition().getName())
+        != DeletionStatus.PENDING_DELETION) {
```

Comment:
suggestion (blocking): use .equals for enums

Reply:
Done.